src/target_intacct/payment_record.py

```diff
@@ -0,0 +1,173 @@
+        config_value = config[name]       
+        if not any(pair[name.upper()] == config_value for pair in ids_list):
+                raise Exception(
+                f"Field {name} with the value {config_value} is missing in Intacct"
+        )        
+
+    # Checks all of the account numbers in the config against the account numbers in Intacct
+    account_numbers = [value for key, value in config.items() if key.startswith('accountno')]
+    for account_no in account_numbers:
+        if not any(int(pair["ACCOUNTNO"]) == account_no for pair in account_ids):
+            raise Exception(
+                f"Field glaccountid with the value {account_no} is missing in Intacct"
+        )        
+
+def payment_record_upload(intacct_client, config) -> None:
+    """Creates payment records in Intacct.
+
+    Retrieves objects from Intacct API for verifying input data
+    Retrieves required data from input
+    Sends entries for uploading to Intacct
+    """
+
+    logger.info("Starting upload.")
+    
+    # Verify config data
+    verify_config_values(intacct_client, config)
+
+    # Get input from pipeline
+    input_value = get_input()
+    
+    # Convert input from dictionary to DataFrames
+    payouts, transactions = process_input(input_value) 
+    grouped_transactions = transactions.groupby('payout_id')
+
+    for payout_id, associated_transactions in grouped_transactions:         
+    
+        # Get all of the reciept totals 
+        gross_amount, total_fees, total_sales_tax = get_receipt_data(associated_transactions) 
+
+        # Get the payout data for the group of transactions
+        payout = payouts.loc[payouts['payout_id'] == payout_id].squeeze()
+        payout_amount = payout["amount"]
+
+        timestamp = payout["available_on"]
+        payout_date = datetime.datetime.fromtimestamp(timestamp)
+        year = payout_date.year
+        month = payout_date.month
+        day = payout_date.day
+        
+        # If payout was negative (more was refunded/in fees than profit made) send to manual payments, otherwise send data to other receipts
+        if(payout_amount > 0):
+            data = {
+                    "paymentdate": get_date_lines(year, month, day),
+                    "payee": config["source"],
+                    "receiveddate": get_date_lines(year, month, day),
+                    "paymentmethod": config["paymentmethod"],
+                    "bankaccountid": config["bankaccountid"], # ENV var
+                    "depositdate": get_date_lines(year, month, day),
+                    "description": config["description"],
+                    "receiptitems": {"lineitem": build_line_items(gross_amount, total_fees, total_sales_tax, config)}}
+            intacct_client.post_other_receipt(data)
+        else:
+            data = {
+                    "bankaccountid": config["bankaccountid"],
```

Comment:
would recommend having the post_other_receipt field order match that of post_manual_payment.  This way it becomes easier to see what are the differences.  For eample if all the fields were sorted alphabetially in order, it would be easy to scan.

Reply:
this is unfortunately the order that is required by Intacct. I agree that it would be nice if they standardized it more :/